src/dbus/item.rs

```diff
@@ -128,7 +131,16 @@ impl<'a> Item<'a> {
             Err(Error::Deleted)
         } else {
             let secret = self.inner.secret(&self.session).await?;
-            Ok(secret.value)
+
+            let value = match self.algorithm {
+                Algorithm::Plain => secret.value,
+                Algorithm::Encrypted => {
+                    let iv = secret.parameters;
+                    let aes_key = self.aes_key.as_ref().unwrap();
```

Comment:
Should keep a note here why unwrap is safe (because it is encrpyted), same for the decrypt call. Which i believe shouldn't have an unwrap